### PR TITLE
feat(#major); friend-tech; add friend-tech subgraph

### PR DESCRIPTION
**Pull request by dhruv-chauhan**
**Description**

- Added friend-tech subgraph using non-standard schema.
- Added configs for networks: base

**Links**

- base: https://okgraph.xyz/?q=dhruv-chauhan%2Ffriend-tech-base 
- alt: https://okgraph.xyz/?q=dhruv-chauhan%2Ffriend-tech-base-2 (fixed: totalPoolCount)

**New Subgraph Checklist**

- [x] Researched protocol / contracts
- [x] Modeled off this [template](https://github.com/messari/subgraphs/tree/master/subgraphs/_reference_) using the latest versions
- [x] Updated title
- [x] Added subgraph(s) to [deployment.json](https://github.com/messari/subgraphs/blob/master/deployment/deployment.json)
- [x] Used standard subgraph libraries (if necessary/available)
- [x] General aggregate metrics can be similar to known sources like the protocol or defi llama (The detailed QA will be done by a protocol specialist)
- [x] Assigned the proper reviewers. If you are unsure, ask in our discord!
- [x] Followed docs, especially [`WALKTHROUGH.md`](https://github.com/messari/subgraphs/blob/master/docs/WALKTHROUGH.md)

### CODE REVIEW

File: subgraphs/friend-tech/src/common/events.ts
Comment (dmelotik) [inline subgraphs/friend-tech/src/common/events.ts:39]: might be cleaner to have a helper functiuon to convert the native amount to the USD amount

File: subgraphs/friend-tech/schema.graphql
Comment (dmelotik) [inline subgraphs/friend-tech/schema.graphql:778]: Could some of these activeX be combined into 1 entity? Like `_ActiveEntity` or will there be conflicts?

File: subgraphs/friend-tech/src/common/metrics.ts
Comment (dmelotik) [inline subgraphs/friend-tech/src/common/metrics.ts:30]: Thoughts on recalculating the USD TVL with the new ETH amounts? That is usually how we do it, but I am not 100% sure if that does/doesn't make sense for friend tech

File: subgraphs/friend-tech/src/common/metrics.ts
Comment (dmelotik) [inline subgraphs/friend-tech/src/common/metrics.ts:99]: can use a helper method for this here too

File: subgraphs/friend-tech/schema.graphql
Comment (dmelotik) [inline subgraphs/friend-tech/schema.graphql:238]: I was thinking ab immutable snapshots, but idk if there enough transactions that it will matter.

The tradeoff is that if there is not a new transaction in a neew day then the previous day snapshot is not created.

File: deployment/deployment.json
Comment (dmelotik) [inline deployment/deployment.json:9770]: turn to prod once QA passes before merging

**Review by melotik — DISMISSED** [review]
code lgtm! Once the QA passes let's change to `prod` and merge :)

File: subgraphs/friend-tech/abis/Shares.json
Comment (melotik) [inline subgraphs/friend-tech/abis/Shares.json:1]: nit: format json

**Review by melotik — APPROVED** [review]
lgtm! We do have a generic sdk, but I will let you decide if it is worth the effort to convert it (prolly not imo, since there are so many new fields, this is alr written, and some of the new fields rely on values used in the sdk)

File: deployment/deployment.json
Comment (melotik) [inline deployment/deployment.json:9772]: change to 2.1.1 (whatever the generic schema is), otherwise it messes up our ingestion

**Comment by melotik** [comment]
https://thegraph.com/hosted-service/subgraph/messari/friend-tech-base

**Pull request merged**
